### PR TITLE
work with additional key values

### PR DESCRIPTION
**Pull request by girazoki**
Add the possibility to inject additional key-values in the sproof-builder that generates the relay root that gets stored in parachain-system.

Rationale:  pallets that verify additional storage items (not those verified by parachain-system) from the relay should be able to proof against the relay root that gets stored in parachain-system. This PR allows to create provide additional nibles that can later be used for verifiability in other pallets

### CODE REVIEW

File: cumulus/primitives/parachain-inherent/src/mock.rs [hunk 1]
```diff
@@ -61,6 +61,8 @@ pub struct MockValidationDataInherentDataProvider<R = ()> {
 	pub raw_downward_messages: Vec<Vec<u8>>,
 	// Inbound Horizontal messages sorted by channel
 	pub raw_horizontal_messages: Vec<(ParaId, Vec<u8>)>,
+	// Additional keys to be injected
```
Comment (bkchr) [inline cumulus/primitives/parachain-inherent/src/mock.rs:64]:
```suggestion
	// Additional keys value pairs that should be injected.
```

**Review by bkchr — APPROVED** [review]

File: cumulus/primitives/parachain-inherent/src/mock.rs (hunk 1 above)
Comment (skunert) [reply cumulus/primitives/parachain-inherent/src/mock.rs]: Typo in the suggestion. Should be
```suggestion
	// Additional key-value pairs that should be injected.
```

**Review by skunert — APPROVED** [review]

File: cumulus/primitives/parachain-inherent/src/mock.rs (hunk 1 above)
Comment (bkchr) [reply cumulus/primitives/parachain-inherent/src/mock.rs]: @girazoki please fix and then we can merge! Ty

Comment (girazoki) [reply cumulus/primitives/parachain-inherent/src/mock.rs]: done!

**Comment by bkchr** [comment]
bot fmt

**Comment by bkchr** [comment]
@girazoki next time please allow maintainers to push changes :P Can you please run rustfmt?

**Comment by girazoki** [comment]
wow somehow this went under my radar. Let me try to do this

**Comment by girazoki** [comment]
@bkchr I think I fixed the fmt issues but I am not sure which nightly version should I use for fmt :(, in any case next time I will open the PR from my personal github to make sure I allow maintaners to push

**Review by ggwpez — APPROVED** [review]

**Pull request merged**